app/views/modals/set-home-page-modal.html

```diff
@@ -0,0 +1,50 @@
+<div class="modal-resource-action">
+  <div class="modal-header">
+    <h1>Set Home Page</h1>
+  </div>
+  <div class="modal-body modal-body-indent">
+    <p>
+      <div class="title">Set your preferred home page.</div>
+      Note: This setting is browser specific and will not be maintained across browsers.
+    </p>
+    <ng-form name="homePageForm">
+      <fieldset ng-disabled="loading">
+        <div class="radio">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="catalog-home">
+            Catalog Home (Default)
+          </label>
+        </div>
+        <div class="radio">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="project-list">
+            My Projects List
+          </label>
+        </div>
+        <div class="radio" ng-show="availableProjects.length >= 1">
+          <label>
+            <input type="radio" ng-model="preferredHomePage" value="project-overview">
+            Overview Page for {{availableProjects.length === 1 ? (selectedProject | displayName) : ':'}}
+          </label>
+          <div class="select-project-container">
+            <select-project ng-if="availableProjects.length > 1"
+                            class="select-project"
+                            on-project-selected="onProjectSelected"
+                            on-open="onOpen"
+                            available-projects="availableProjects"
+                            selected-project="selectedProject"
+                            hide-create-project="true"
+                            hide-label="true">
+            </select-project>
+          </div>
+        </div>
+      </fieldset>
+    </ng-form>
+  </div>
+  <div class="modal-footer">
+      <button class="btn btn-lg btn-primary" type="button"
+              ng-disabled="(preferredHomePage === 'project-overview' && homePageForm.$invalid) || homePageForm.$pristine"
```

Comment:
I think this can just be... ?

```html
ng-disabled="homePageForm.$invalid || homePageForm.$pristine"
```

Reply:
Nevermind I see why it's not possible.

It would be better to add an `ng-required` attribute to the `select-project` component and then use...

```html
<select-project
  ng-required="preferredHomePage === 'project-overview'"
  ....
```

Then you can change this disabled check to the above.

Reply:
Thanks for the tip, I hadn't used `ng-required` before...very useful :-)